code/modules/awaymissions/gateway.dm

```diff
@@ -47,7 +47,10 @@ GLOBAL_DATUM_INIT(the_gateway, /obj/machinery/gateway/centerstation, null)
 /obj/machinery/gateway/centerstation/Initialize()
 	..()
 	update_icon()
-	wait = world.time + config.gateway_delay	//+ thirty minutes default
+	wait = world.time + config.gateway_delay
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/machinery/gateway/centerstation/LateInitialize()
```

Comment:
What's the reason for moving this to late initialize? Was `awaygate` getting set to null because it hadn't spawned in yet properly or something?

Reply:
awaygate should have been defined in Initialize(), but it wasn't because a GLOB it was searching in wasn't defined by that point. AA nicely found that moving it to LateInitialize() made it work properly.

Reply:
Yeah sounds good to me.

Reply:
@SteelSlayer The away gate gets added to GLOB.machines during its own Initialize() call, which can cause issues if the station gate fires off before that one does. By moving the search to LateInitialize(), an away gate will always be found since there's no longer a race condition.